src/Versioned.php

```diff
@@ -2023,62 +2009,12 @@ public function stagesDifferRecursive(): bool
         return $service->stagesDifferRecursive($this->owner);
     }
 
-    /**
-     * @param string $filter
-     * @param string $sort
-     * @param string $limit
-     * @return ArrayList<Versioned_Version>
-     */
     public function Versions($filter = "", $sort = "", $limit = "", $join = "")
     {
-        $owner = $this->owner;
-
-        // When an object is not yet in the Database, we can't get its versions
-        if (!$owner->isInDB()) {
-            return ArrayList::create();
-        }
-
-        // Make sure the table names are not postfixed (e.g. _Live)
-        $oldMode = static::get_reading_mode();
-        static::set_stage(static::DRAFT);
-
-        $list = DataObject::get(DataObject::getSchema()->baseDataClass($owner), $filter, $sort, $limit);
-
-        $query = $list->dataQuery()->query();
-
-        $baseTable = null;
-        foreach ($query->getFrom() as $table => $tableJoin) {
-            if (is_string($tableJoin) && $tableJoin[0] == '"') {
-                $baseTable = str_replace('"', '', $tableJoin ?? '');
-            } elseif (is_string($tableJoin) && substr($tableJoin ?? '', 0, 5) != 'INNER') {
-                $query->setFrom([
-                    $table => "LEFT JOIN \"$table\" ON \"$table\".\"RecordID\"=\"{$baseTable}_Versions\".\"RecordID\""
-                        . " AND \"$table\".\"Version\" = \"{$baseTable}_Versions\".\"Version\""
-                ]);
-            }
-            $query->renameTable($table, $table . '_Versions');
-        }
-
-        // Add all <basetable>_Versions columns
-        foreach (Config::inst()->get(static::class, 'db_for_versions_table') as $name => $type) {
-            $query->selectField(sprintf('"%s_Versions"."%s"', $baseTable, $name), $name);
-        }
-
-        $query->addWhere([
-            "\"{$baseTable}_Versions\".\"RecordID\" = ?" => $owner->ID
-        ]);
-        $query->setOrderBy(($sort) ? $sort
-            : "\"{$baseTable}_Versions\".\"LastEdited\" DESC, \"{$baseTable}_Versions\".\"Version\" DESC");
-
-        $records = $query->execute();
-        $versions = new ArrayList();
-
-        foreach ($records as $record) {
-            $versions->push(new Versioned_Version($record));
-        }
-
-        Versioned::set_reading_mode($oldMode);
-        return $versions;
+        $id = $this->getOwner()->ID;
+        $owner = $this->getOwner();
```

Comment:
```suggestion
        $owner = $this->getOwner();
        $id = $owner->ID;
```
Tiny subjective change but it's nicer to not call that method a second time if we're making a variable of it anyway.

Reply:
Updated